app/models/restaurant.rb

```diff
@@ -19,10 +19,10 @@ class Restaurant < ApplicationRecord
   validates_inclusion_of :accepts_10bis, in: [true, false]
 
   belongs_to :cuisine
-  has_many :reviews
+  has_many :reviews, dependent: :destroy
 
 
   def calc_rating
-    self.rating = reviews.empty? ? nil: reviews.average(:rating)
+    self.rating = reviews.present? && reviews.average(:rating) ? reviews.average(:rating).round : nil
```

Comment:
in ruby there is the `safe navigation` operator: `&.`
you can use it
`reviews&.average(:ratings).round`

which returns nil or the result.

Reply:
Cool! Thanks.